week2/homework/src/index.js

```diff
@@ -1,3 +1,134 @@
 'use strict';
+const fs = require('fs');
+const command = process.argv[2];
+const commandFirstItem = process.argv[3];
+const newData = process.argv[4];
 
-// TODO: Write the homework code in this file
+function help() {
+  fs.readFile('./help.txt', 'utf8', function (error, data) {
+    if (error) {
+      console.log(error);
+    }
+    else {
+      console.log(data);
+    }
+  });
+};
+
+function list() {
+  fs.readFile('./toDo.json', 'utf8', function (error, data) {
+    if (error) {
+      console.log(error);
+    }
+    else {
+      const myToDoList = JSON.parse(data);
+      if (Object.values(myToDoList).length === 0) {
+        console.log('You have not any plan in your To Do List');
+      }
+      else {
+        console.log(myToDoList);
+      }
+    }
+  });
+};
+
+function add(commandFirstItem) {
+  fs.readFile('./toDo.json', 'utf8', function (error, data) {
+    if (error) {
+      console.log(error);
+    }
+    else {
+      const myToDoList = JSON.parse(data);
+      const index = Object.keys(myToDoList).length + 1;
+      myToDoList[index] = commandFirstItem;
+      fs.writeFile(
+        './toDo.json',
+        JSON.stringify(myToDoList),
+        function (error) {
+          if (error) {
+            console.error(error);
+          }
+        }
+      );
+    }
+  });
+}
+
+function remove(commandFirstItem) {
+  fs.readFile('./toDo.json', 'utf8', function (error, data) {
+    if (error) {
+      console.log(error);
+    }
+    else {
+      const myToDoList = JSON.parse(data);
+      if (commandFirstItem >= 1) {
+        delete myToDoList[parseInt(commandFirstItem)];
+        fs.writeFile(
+          './toDo.json',
+          JSON.stringify(myToDoList),
+          (error) => {
+            if (error) { console.log(error); }
+          });
+      }
+      else {
+        console.log('The parameter you entered as index is invalid');
+      }
+    }
+  });
+}
+
+function reset() {
+  fs.writeFile(
+    './toDo.json',
+    '{}',
+    (error) => {
+      if (error) { console.log(error); }
+    }
+  );
+};
+
+function update(commandFirstItem, newData) {
+  fs.readFile('./toDo.json', 'utf8', function (error, data) {
+    if (error) {
+      console.log(error);
+    }
+    else {
+      const myToDoList = JSON.parse(data);
+      if (commandFirstItem > 0) {
```

Comment:
Here you can apply the early return again.
The trick is: instead of thinking if `commandFirstItem` is what I want then do this, you need to think: if `commandFirstItem` is **not** what i want then stop and return an error :)

Does that makes sense?

Reply:
Yes I tried to edit it.